SPEC.md

```diff
@@ -1,12 +1,46 @@
 # Code Climate Engine Specification
+## Abstract
```

Comment:
💄 we generally do blank lines between consecutive headers in markdown, I think. (`markdownlint` might help enforce this? Might not be worth going down the "add cc engines config" path here right now, though.)